src/tree_schema.h

```diff
@@ -2392,6 +2392,7 @@ struct lys_module {
     ly_bool implemented;             /**< flag if the module is implemented, not just imported */
     ly_bool to_compile;              /**< flag marking a module that was changed but not (re)compiled, see
                                           ::LY_CTX_EXPLICIT_COMPILE. */
+    ly_bool internal;                /**< flag indicating that this is an automatically loaded internal module */
```

Comment:
This will not be accepted because you are changing the ABI. But I do not understand what it is needed for, anyway. Why do you want to skip internal modules in the tree output?

Reply:
I don't have a strong opinion on this and don't mind removing it.

The reason I filtered these out to begin with was because it looked to me like yanglint was reporting on libyang implementation details.  The internal modules appear in the diagram even if they are not specified in the module-set instance for the mount point.  Some of the modules are generally necessary, but for example, libyang loads ietf-yang-schema-mount even if none of the modules in the context has a mount point.  I wasn't sure that it really makes sense to display this.

Reply:
I think it is because you can always instantiate the internal modules data in the mounted data so they technically are always part of the yang-library data.